modules/govuk_data_sync_in_progress/manifests/init.pp

```diff
@@ -0,0 +1,69 @@
+# == Class: govuk_data_sync_in_progress
+#
+# Configure the `data_sync_in_progress` fact and optionally allows commands
+# to run when the data sync starts and finishes.
+#
+# === Parameters
+#
+# [*start_command*]
+#   Command to run when the data sync starts.
+#
+# [*finish_command*]
+#   Command to run when the data sync finishes.
+#
+define govuk_data_sync_in_progress(
+  $start_command = undef,
+  $finish_command = undef
+) {
+  $fact_path = '/etc/govuk/env.d/FACTER_data_sync_in_progress'
+
+  $start_hour = 22
+  $start_minute = 0
+
+  $finish_hour = 5
+  $finish_minute = 45
+
+  if !defined(File[$fact_path]) {
```

Comment:
I don't really get this conditional, what is it doing?

Reply:
It’s so that this definition is generic enough that it can be used multiple times on the same machine class without Puppet complaining that there is already a definition for that file.

I don’t know if it’s a problem at the moment (since Smokey and Gor run on different machine classes I think) but I wanted to make the definition generic so it didn’t confuse anyone in the future.

Reply:
Ok, that makes sense :+1: